src/modules/hotel/hotel.controller.ts

```diff
@@ -0,0 +1,88 @@
+import { FastifyReply, FastifyRequest } from "fastify";
+import {
+    createHotel,
+    browseHotel,
+    showHotel,
+    updateHotel,
+    deleteHotel
+} from "./hotel.service";
+import {
+    CreateHotelInput,
+    UpdateHotelInput,
+    ShowHotelParams,
+    DeleteHotelParams
+} from "./hotel.schema";
+import { error_message } from "./hotel.errors";
+
+export async function createHotelHandler(
+    request: FastifyRequest<{
+        Body: CreateHotelInput;
+    }>,
+    reply: FastifyReply
+) {
+    try {
+        const hotel = await createHotel(request.body);
+
+        reply.code(201).send(hotel);
+    } catch (e) {
+        return reply.badRequest(await error_message(e));
+    }
+}
+
+export async function browseHotelHandler(
+    request: FastifyRequest,
+    reply: FastifyReply
+) {
+    try {
+        const hotels = await browseHotel();
```

Comment:
Should be cached with redis

Reply:
All gets should be cached, and invalidated on update and delete

Reply:
Changes added